src/server/frontend_wayland/wl_surface.cpp

```diff
@@ -142,7 +143,7 @@ void mf::WlSurface::attach(std::experimental::optional<wl_resource*> const& buff
 
     role->visiblity(!!buffer);
 
-    pending.buffer = buffer.value_or(nullptr);
+    pending.buffer = buffer;
```

Comment:
If a null buffer is sent, it will come through as a nullopt, but nullopt means something different in `pending` (no buffer at all), so I believe the original was correct, unless there is something I'm missing. See https://github.com/MirServer/mir/blob/6dd056a4277b79930bb6b92d16e69d0300c6af8e/src/server/frontend_wayland/wl_surface.h#L64-L68

Reply:
Ack. My mistake.